packages/eslint-plugin/src/rules/no-magic-numbers.ts

```diff
@@ -1,16 +1,30 @@
 import {
-  TSESTree,
   AST_NODE_TYPES,
+  TSESTree,
 } from '@typescript-eslint/experimental-utils';
 import baseRule from 'eslint/lib/rules/no-magic-numbers';
 import * as util from '../util';
 
 type Options = util.InferOptionsTypeFromRule<typeof baseRule>;
 type MessageIds = util.InferMessageIdsTypeFromRule<typeof baseRule>;
 
-const baseRuleSchema = Array.isArray(baseRule.meta.schema)
-  ? baseRule.meta.schema[0]
-  : baseRule.meta.schema;
+// Extend base schema with additional property to ignore TS numeric literal types
+const schema = util.deepMerge(
```

Comment:
I had to do this in order to make it work since now `schema` accepts a `readonly JSONSchema4[]`, the linter started to complain about unsafe assignment... but it seems that it's the right way to do it as per looking to the other existent code in the codebase. In any case, let me know if it's desirable or not, so I can address that accordingly.

Reply:
> the linter started to complain about unsafe assignment

This was more than likely a red-herring that occurred due to out-of-date/weird types.
I'm happy with this approach however